suddenly started having error TS2305: Module '"../../node_modules/ng2-nouislider/ng2-nouislider"' has no exported member 'NouisliderModule'.

Suddenly stated having error TS2305: Module '"../../node_modules/ng2-nouislider/ng2-nouislider"' has no exported member 'NouisliderModule'. 

can't build angular project anymore! please help

--- a/demo/app.component.ts
+++ b/demo/app.component.ts
@@ -2,7 +2,7 @@ import { Component, OnInit } from '@angular/core';
 import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
 import './app.component.scss';
 
-import { NouiFormatter } from '../src/ng2-nouislider';
+import { NouiFormatter } from '../src/public_api';
 
 export class TimeFormatter implements NouiFormatter {
   to(value: number): string {

--- a/demo/app.module.ts
+++ b/demo/app.module.ts
@@ -3,7 +3,7 @@ import { BrowserModule } from '@angular/platform-browser';
 import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 
 import { AppComponent } from './app.component';
-import { NouisliderModule } from '../src/ng2-nouislider';
+import { NouisliderModule } from '../src/public_api';
 
 @NgModule({
   imports: [

--- a/ng-package.json
+++ b/ng-package.json
@@ -1,6 +1,9 @@
 {
   "$schema": "./node_modules/ng-packagr/ng-package.schema.json",
   "lib": {
-    "entryFile": "src/ng2-nouislider.ts"
+    "entryFile": "src/public_api.ts",
+    "externals": {
+      "nouislider": "noUiSlider"
+    }
   }
 }

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "Angular2 noUiSlider component",
   "scripts": {
     "contributors": "all-contributors generate",
-    "lint": "tslint src/ng2-nouislider.ts",
+    "lint": "tslint src/public_api.ts",
     "build": "npm run lint && ng-packagr -p ng-package.json",
     "start": "npm run build && webpack-dev-server --host 0.0.0.0",
     "deploy": "NODE_ENV=prod npm run build && NODE_ENV=prod webpack && gh-pages -d demo",

--- a/src/ng2-nouislider.component.ts
+++ b/src/ng2-nouislider.component.ts
@@ -8,7 +8,6 @@ import {
   OnInit,
   OnChanges,
   Output,
-  NgModule,
   Renderer2
 } from '@angular/core';
 import {
@@ -264,10 +263,3 @@ export class NouisliderComponent implements ControlValueAccessor, OnInit, OnChan
   }
 }
 
-
-@NgModule({
-  imports: [],
-  exports: [NouisliderComponent],
-  declarations: [NouisliderComponent],
-})
-export class NouisliderModule { }

--- a/src/ng2-nouislider.module.ts
+++ b/src/ng2-nouislider.module.ts
@@ -1,0 +1,8 @@
+import {NgModule} from "@angular/core";
+import {NouisliderComponent} from "./ng2-nouislider.component";
+
+@NgModule({
+    exports: [NouisliderComponent],
+    declarations: [NouisliderComponent],
+})
+export class NouisliderModule { }

--- a/src/public_api.ts
+++ b/src/public_api.ts
@@ -1,0 +1,2 @@
+export * from './ng2-nouislider.component';
+export * from './ng2-nouislider.module';

--- a/test/ng2-nouislider.spec.ts
+++ b/test/ng2-nouislider.spec.ts
@@ -4,7 +4,7 @@ import { By } from '@angular/platform-browser';
 import { Component, DebugElement } from '@angular/core';
 import { FormsModule, FormBuilder, FormGroup, ReactiveFormsModule } from '@angular/forms';
 
-import { DefaultFormatter, NouisliderModule, NouisliderComponent } from '../src/ng2-nouislider.ts';
+import { DefaultFormatter, NouisliderModule, NouisliderComponent } from '../src/public_api';
 
 describe('Default Formatter', () => {
   let formatter: DefaultFormatter;

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -25,7 +25,7 @@
     "node_modules"
   ],
   "files": [
-    "src/ng2-nouislider.ts"
+    "src/public_api.ts"
   ],
   "angularCompilerOptions": {
     "genDir": "compiled"

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -52,7 +52,7 @@ module.exports = function makeWebpackConfig() {
     // only discover files that have those extensions
     extensions: ['', '.ts', '.js', '.json', '.css', '.scss', '.html'],
     alias: {
-      'ng2nouislider': path.resolve('./src/ng2-nouislider.ts')
+      'ng2nouislider': path.resolve('./src/public_api.ts')
     }
   };
 

